pkg/operation/botanist/monitoring.go

```diff
@@ -413,47 +418,149 @@ func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
-		{"", "v1", "ConfigMap", "prometheus-config"},
-		{"", "v1", "ConfigMap", "prometheus-rules"},
-		{"", "v1", "ConfigMap", "blackbox-exporter-config-prometheus"},
-		{"", "v1", "Secret", "prometheus-basic-auth"},
-		{"extensions", "v1beta1", "Ingress", "prometheus"},
-		{"networking", "v1", "Ingress", "prometheus"},
-		{"autoscaling.k8s.io", "v1beta2", "VerticalPodAutoscaler", "prometheus-vpa"},
-		{"", "v1", "ServiceAccount", "prometheus"},
-		{"", "v1", "Service", "prometheus"},
-		{"", "v1", "Service", "prometheus-web"},
-		{"apps", "v1", "StatefulSet", "prometheus"},
-		{"rbac", "v1", "ClusterRoleBinding", "prometheus-" + b.Shoot.SeedNamespace},
-		{"", "v1", "PersistentVolumeClaim", "prometheus-db-prometheus-0"},
+	for _, obj := range []runtime.Object{
+		&corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics-seed",
+			},
+		},
+		&rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics-seed",
+			},
+		},
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics-seed",
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics-seed",
+			},
+		},
+		&autoscalingv1beta2.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics-seed-vpa",
+			},
+		},
+
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics",
+			},
+		},
+		&autoscalingv1beta2.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics-vpa",
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics",
+			},
+		},
+
+		&networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "allow-from-prometheus",
+			},
+		},
+		&networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "allow-prometheus",
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "prometheus-config",
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "prometheus-rules",
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "blackbox-exporter-config-prometheus",
+			},
+		},
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "prometheus-basic-auth",
+			},
+		},
+		&extensionsv1beta1.Ingress{
```

Comment:
I know it is not introduced with this PR but why the same resource is deleted 2 time from different apiVersion. Ingress will be no longer served from `extensions/v1beta1` starting v1.22 according to the current plans and this is far in the future. I don't see why we need deletion also from `networking.k8s.io/v1beta1`.

Reply:
Good point, removed `networkingv1beta1.Ingress`.